budget_allocation_dimension/models/base_budget_move.py

```diff
@@ -16,11 +16,40 @@ def _get_dimension_fields(self):
             x for x in self.fields_get().keys() if x.startswith("x_dimension_")
         ]
 
-    def _where_query_source_fund(self, docline):
-        where_query = super()._where_query_source_fund(docline)
+    def _get_condition_skip_dimension(self):
+        """
+        This function will check state 2 module
+            1. budget_source_fund
+            2. budget_allocation_dimension_fund
+
+        ============ budget.monitor.report ============
+        - Not install module `budget_source_fund` -> query dimension
+
+        ============ source.fund.monitor.report ============
+        - Install module `budget_source_fund`,
+          Not install module `budget_allocation_dimension_fund` -> skip dimension
+        - Install module `budget_source_fund`,
+          install module `budget_allocation_dimension_fund` -> query dimension
+        """
+        dimension_fund = self.env["ir.module.module"].search(
+            [("name", "=", "budget_allocation_dimension_fund")]
+        )
+        # Skip query dimension
+        if (
+            self.env.get("source.fund.monitor.report", "/") != "/"
```

Comment:
Are there better way to check installed module?

Reply:
ตอนนี้ในระบบมีหลาย module และแยกจากกันหลายส่วน อาจจะ install หรือไม่ก็ได้ เช่น
1. กลุ่ม budget_allcoation
    - budget_allocation
    - budget_allocation_dimension
    - budget_allocation_dimension_fund
2. budget_constraint
3. budget_source_fund

ซึ่งถ้ามีการ install module budget_allocation + budget_constraint ระบบจะยังไม่มี source fund monitoring และทำให้เกิด error ได้ เลยต้องมีการตรวจสอบก่อนว่ามีการ install module budget_source_fund ไปหรือยัง